### ISSUE TITLE
`create_model` behaves strangely for required fields with `Field` annotations

### ISSUE THREAD

**Issue by rwb27** [pydantic/pydantic#7064]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

When I create a model dynamically using `create_model`, required fields annotated with `Field` show as optional, with a default value of `Ellipsis`.  Specifically, passing `(<type>, ...)` is OK when `<type>` is either a simple type, or an `Annotated` with multiple fields in it - but for the specific case of `Annotated[<type>, Field(<arguments>)]` it goes wrong.

The symptom is that the field is created, with the correct type but with `required=False` and `default=Ellipsis`, instead of `required=True` and no default.

I believe I have found a work-around: simply using `Field(...)` instead of `...` seems to fix the problem.

The same thing occurs if I call `FieldInfo.from_annotated_attribute()` directly - the four test cases below do exactly the same thing (i.e. `baz` fails but the other three work).

### Example Code

```Python
from pydantic import create_model, Field
from typing import Annotated

m = create_model(
    "test_model",
    foo=(int, ...),  # a required integer field - works fine
    bar=(Annotated[int, Field(gt=0), Field(lt=10)], ...),  # this works too
    baz=(Annotated[int, Field(gt=0)], ...),  # a required integer >0, fails
    fixed=(Annotated[int, Field(gt=0)], Field(...)),  # works
)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.1.1
        pydantic-core version: 2.4.0
          pydantic-core build: profile=release pgo=false mimalloc=true
                 install path: C:\Users\Richard\repos\my_projects\labthings-fastapi\.venv\Lib\site-packages\pydantic
               python version: 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]     
                     platform: Windows-10-10.0.19045-SP0
     optional deps. installed: ['email-validator', 'typing-extensions']
```

Selected Assignee: @lig

**Comment by rwb27** [comment]
This may well be related to #2248.

**Comment by lig** [comment]
It looks like this one was fixed and cannot be reproduced in Pydantic 2.3.0.

This test fully passes in Pydantic 2.3.0:

```py
import pytest

def test_7064():
    from typing import Annotated

    from pydantic import Field, create_model

    Model = create_model(
        'test_model',
        foo=(int, ...),  # a required integer field - works fine
        bar=(Annotated[int, Field(gt=0), Field(lt=10)], ...),  # this works too
        baz=(Annotated[int, Field(gt=0)], ...),  # a required integer >0, fails
        fixed=(Annotated[int, Field(gt=0)], Field(...)),  # works
    )

    assert Model(foo=0, bar=1, baz=2, fixed=3).model_dump() == {'foo': 0, 'bar': 1, 'baz': 2, 'fixed': 3}

    with pytest.raises(ValidationError, match='bar'):
        Model(foo=0, bar=-1, baz=2, fixed=3)
    with pytest.raises(ValidationError, match='bar'):
        Model(foo=0, bar=11, baz=2, fixed=3)
    with pytest.raises(ValidationError, match='baz'):
        Model(foo=0, bar=1, baz=-1, fixed=3)
    with pytest.raises(ValidationError, match='fixed'):
        Model(foo=0, bar=1, baz=2, fixed=-1)
```
